sycl/include/CL/sycl/ONEAPI/group_algorithm.hpp

```diff
@@ -485,11 +473,13 @@ EnableIfIsNonNativeOp<T, BinaryOperation> reduce(Group g, T x,
 }
 
 template <typename Group, typename V, typename T, class BinaryOperation>
-EnableIfIsScalarArithmeticNativeOp<T, BinaryOperation>
+detail::enable_if_t<(detail::is_generic_group<Group>::value &&
+                     detail::is_scalar_arithmetic<V>::value &&
```

Comment:
This seems clearly wrong. I'll add details in a few hours.

Reply:
@rolandschulz and I spoke about this offline.

The `is_scalar_arithmetic` traits here are checking that each of the input types are SYCL fundamental types.  This is all that is required by the specification, and is sufficient because the fundamental types are all closed under the operations defined by the native function objects.  If these functions are later extended to support more general combinations of types and operations, the traits may have to change -- for example, we may need to consider whether certain types are convertible to fundamental types, or that the return value of the binary operator applied to the input types is a fundamental type.